File: SDKs/googleapis.com/sql/v1beta4/typescript/src/sdk/models/shared/tier.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// Tier
/** 
 * A Google Cloud SQL service tier resource.
**/
export class Tier extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=DiskQuota" })
  diskQuota?: string;

  @SpeakeasyMetadata({ data: "json, name=RAM" })
  ram?: string;

  @SpeakeasyMetadata({ data: "json, name=kind" })
  kind?: string;

  @SpeakeasyMetadata({ data: "json, name=region" })
  region?: string[];

  @SpeakeasyMetadata({ data: "json, name=tier" })
  tier?: string;
}

File: SDKs/googleapis.com/sql/v1beta4/python/src/sdk/models/shared/tier.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class Tier:
    r"""Tier
    A Google Cloud SQL service tier resource.
    """
    
    disk_quota: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DiskQuota') }})
    ram: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('RAM') }})
    kind: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('kind') }})
    region: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('region') }})
    tier: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tier') }})
    
